shims/common/src/main/scala/io/glutenproject/extension/JoinSelectionOverrideShim.scala

```diff
@@ -137,7 +134,9 @@ class JoinSelectionOverrideShim(val backendLib: BackendLib)
     joinType match {
       case _: InnerLike | RightOuter | FullOuter => true
       // For Velox backend, build right and left are both supported for LeftOuter and LeftSemi.
-      case LeftOuter | LeftSemi => backendLib.equals(BackendLib.VELOX)
+      case LeftOuter | LeftSemi =>
+        GlutenConfig.getConf.isVeloxBackend ||
+        GlutenConfig.getConf.isGazelleBackend
```

Comment:
With this change, maybe we need to check if gazelle backend aready supports build left for left outer and build right for right outer.

Reply:
I didn't dive into the concrete join code. The change is to keep the same behavior as current - From current code `backendLib.equals(BackendLib.VELOX)`, actually it implies `backendLib.equals(BackendLib.VELOX || BackendLib.GAZELLE)` since Gazelle backend is reusing Velox backend's name. In the patch I made the code clearer by giving Gazelle its own name.

Reply:
So if the code was specific for Velox BE (was it?), I can remove condition `GlutenConfig.getConf.isGazelleBackend` then we can have another ticket to check if that should be added later on.

Reply:
Thanks for your explain. This part of code was specific for Velox backend, and I did not check that for gazelle backend. Double-check may be needed if we want to support this behavior for gazelle backend.

Reply:
Fair enough. I am removing the condition for gazelle be.